validator/decorator.py

```diff
@@ -28,6 +34,29 @@ def wrap(function):
     return wrap
 
 
+def register_cleanup(cleanup):
+    """Register a cleanup function to be called at the end of every validation
+    task. Takes either a callable, or a class with a `cleanup` class method."""
+
+    if not callable(cleanup):
```

Comment:
Yeah, I considered that, but the docstring says `callable`, and I figured that anyone writing a callable class would think of that possibility. I guess it couldn't hurt to be extra specific, though.